### ISSUE TITLE
Can't apply label weight to loss function

### ISSUE THREAD

**Issue by SidneyLann** [deeplearning4j/deeplearning4j#9149]
#### Issue Description

Please describe our issue, along with:
- expected behavior
sameDiff.loss.softmaxCrossEntropy(“loss”, label=[1024,6], out=[1024,6], weight=[6])
Can apply label weight to loss function like above line.

- encountered behavior
 Failed to execute op

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version beta7
* Platform information (OS, etc) 
* CUDA version, if used no
* NVIDIA driver version, if in use

#### Additional Information

Where applicable, please also provide:

* Full log or exception stack trace (ideally in a Gist: gist.github.com)
* pom.xml file or similar (also in a Gist)
[main] ERROR org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner - Failed to execute op softmax_cross_entropy_loss. Attempted to execute with 3 inputs, 1 outputs, 1 targs,0 bargs and 1 iargs. Inputs: [(FLOAT,[1024,6],c), (FLOAT,[6],c), (FLOAT,[1024,6],c)]. Outputs: [(FLOAT,,c)]. tArgs: [0.0]. iArgs: [3]. bArgs: -. Input var names: [out, labelWeight, label]. Output var names: [loss] - Please see above message (printed out from c++) for a possible cause of error.
java.lang.RuntimeException: NDArray::tile method - shapeInfo of target array is not suitable for tile operation !
at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:1924)
at org.nd4j.linalg.factory.Nd4j.exec(Nd4j.java:6575)
at org.nd4j.autodiff.samediff.internal.InferenceSession.doExec(InferenceSession.java:495)
at org.nd4j.autodiff.samediff.internal.InferenceSession.getOutputs(InferenceSession.java:222)
at org.nd4j.autodiff.samediff.internal.TrainingSession.getOutputs(TrainingSession.java:152)
at org.nd4j.autodiff.samediff.internal.InferenceSession.getOutputs(InferenceSession.java:68)
at org.nd4j.autodiff.samediff.internal.AbstractSession.output(AbstractSession.java:411)
at org.nd4j.autodiff.samediff.internal.TrainingSession.trainingIteration(TrainingSession.java:118)
#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!

**Comment by jljljl** [comment]
@SidneyLann "Input var names: [out, labelWeight, label]" but you use "label=[1024,6], out=[1024,6], weight=[6]". Also you can use labels masking.

**Comment by jljljl** [comment]
I don't know arguments ordering but you can check "Inputs: [(FLOAT,[1024,6],c), (FLOAT,[6],c), (FLOAT,[1024,6],c)]", or check message from C++.

**Comment by SidneyLann** [comment]
There are 6 combinations for label=[1024,6], out=[1024,6], weight=[6], no one has the success running.

**Comment by jljljl** [comment]
@SidneyLann Check C++ log and try masking.

**Comment by jljljl** [comment]
See "NDArray::tile method - shapeInfo of target array is not suitable for tile operation !"

**Comment by SidneyLann** [comment]
Why need masking?   label=[1024,6], out=[1024,6], weight=[6] should be applied label weight to loss function successfully. It may be the problem in DL4J itself?

**Comment by SidneyLann** [comment]
This issue still exists in M1, had anybody used label weight in softmaxCrossEntropy(String name, SDVariable oneHotLabels,
      SDVariable logitPredictions, SDVariable weights) successfully?

**Comment by agibsonccc** [comment]
Yes not all issues people had were fixed in M1. It's been 1 year since
release.  Patch notes plus new docs and examples will be published soon.
There will be a quick follow up release with bug fixes based on user
feedback and module priority. Samediff is high up to do a pass. I expect to
do that in a few weeks.

**Comment by SidneyLann** [comment]
Ok, thanks!

**Comment by agibsonccc** [comment]
@SidneyLann  looked in to this, in M1 and snapshots you use:         sameDiff.loss.weightedCrossEntropyWithLogits()
closing this

**Comment by SidneyLann** [comment]
java.lang.UnsupportedOperationException: Please extend DynamicCustomOp.doDiff to support SameDiff backprop operations. Op: org.nd4j.linalg.api.ops.impl.loss.WeightedCrossEntropyLoss


can't train.

**Comment by SidneyLann** [comment]
@agibsonccc Could you reopen this issue?

**Comment by agibsonccc** [comment]
@SidneyLann could you file a separate issue? Thanks!
